src/client/app/shared/canvas/canvas.component.ts

```diff
@@ -18,37 +18,53 @@ export class CanvasComponent {
   panning: boolean = false;
   // The last point seen during the pan that is currently in progress.
   lastPanPnt: Point = null; //TODO(eyuelt): make this nullable after TS2 update
+  // Returns the bounding box of the canvas.
+  getBounds: {():ClientRect} = null;
 
-  constructor(private canvasService: CanvasService) {
+  // Note: ElementRef should be treated as read-only to avoid XSS vulnerabilites.
+  constructor(elementRef: ElementRef, private canvasService: CanvasService) {
+    this.getBounds = () => { return elementRef.nativeElement.getBoundingClientRect(); };
+    this.canvasService.setCanvasBoundsGetter(this.getBounds);
   }
 
   // TODO(eyuelt): should this use HostListener or the template event binding?
   onMousewheel(event: WheelEvent) {
-    const zoomScale = 1 + (event.deltaY * -0.01);
-    const zoomPnt = {x: event.offsetX, y: event.offsetY};
-    this.canvasService.zoom(zoomPnt, zoomScale);
     event.stopPropagation();
     event.preventDefault();
+    const zoomScale = 1 + (event.deltaY * -0.01);
+    const zoomPnt = {x: event.clientX - this.getBounds().left, y: event.clientY - this.getBounds().top};
```

Comment:
Try to stick to 80 character line length, as per Google JS Style Guide?
https://google.github.io/styleguide/javascriptguide.xml